### PR TITLE
Add instruction for BWC tests on distribution level

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
Adding onboarding instruction and documents for BWC tests on distribution level for plugins.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-build/issues/2870

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ Each plugin can test various functionalities in their bwc tests and add more sce
 
 The BWC tests running on distribution level are using the same framework from OpenSearch. The test cluster is spin up with the `latest` distribution bundle of provided version exclusively when the project is initialized with property `-PcustomDistributionDownloadType=bundle`.
 
-The distribution bundle will be obtained with this format URL: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/__REPLACE_VERSION__/latest/linux/x64/tar/dist/opensearch/opensearch-__REPLACE_VERSION__-linux-x64.tar.gz.
+The distribution bundle will be obtained with this format URL: `https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/__REPLACE_VERSION__/latest/linux/x64/tar/dist/opensearch/opensearch-__REPLACE_VERSION__-linux-x64.tar.gz`.
```
Comment (saratvemulapalli) [inline TESTING.md:105]: An example would be nice.

Comment (zelinh) [reply TESTING.md]: @saratvemulapalli Updated with the example for v1.3.2. Thanks.

**Review by saratvemulapalli — DISMISSED** [review]

File: TESTING.md [hunk 2]
```diff
@@ -98,6 +98,17 @@ See [anomaly-detection#158](https://github.com/opensearch-project/anomaly-detect
 
 Each plugin can test various functionalities in their bwc tests and add more scenarios if needed.
 
+#### BWC tests on distribution bundle level
```
Comment (dblock) [inline TESTING.md:101]: TOC needs updating, please?

**Review by dblock — CHANGES_REQUESTED** [review]

File: TESTING.md (hunk 2 above)
Comment (zelinh) [reply TESTING.md:102]: @dblock Thanks dB. Just updated the table. Thanks!

**Review by VachaShah — APPROVED** [review]

**Comment by zelinh** [comment]
@dblock @saratvemulapalli Could you help review another times. Need one more approval. Thanks.

**Review by saratvemulapalli — APPROVED** [review]

**Comment by zelinh** [comment]
Merging this for now.

**Pull request merged**